client/service_update.go

```diff
@@ -15,13 +15,21 @@ func (cli *Client) ServiceUpdate(ctx context.Context, serviceID string, version
 	var (
 		headers map[string][]string
 		query   = url.Values{}
+		distErr error
 	)
 
 	if options.EncodedRegistryAuth != "" {
 		headers = map[string][]string{
 			"X-Registry-Auth": {options.EncodedRegistryAuth},
+			"version":         {cli.version},
+		}
+	} else {
+		headers = map[string][]string{
+			"version": {cli.version},
 		}
 	}
+	// update CLI version
+	headers["version"] = []string{cli.version}
```

Comment:
`headers["version"]` was already set above. Suggest initializing the map with `version` already set, and simplifying the conditionals above.